crates/apps/src/lib/cli/client.rs

```diff
@@ -298,6 +300,42 @@ impl CliApi {
                         tx::submit_validator_metadata_change(&namada, args)
                             .await?;
                     }
+                    Sub::ShieldedSync(ShieldedSync(mut args)) => {
+                        let client = client.unwrap_or_else(|| {
+                            C::from_tendermint_address(&mut args.ledger_address)
+                        });
+                        client.wait_until_node_is_synced(&io).await?;
+                        let args = args.to_sdk(&mut ctx);
+                        let namada = ctx.to_sdk(client, io);
+                        let wallet_keys: Vec<ExtendedViewingKey> = namada
+                            .wallet()
+                            .await
+                            .get_viewing_keys()
+                            .values()
+                            .copied()
+                            .collect();
+                        let vks = args
+                            .viewing_keys
+                            .into_iter()
+                            .map(|vk| ExtendedFullViewingKey::from(vk).fvk.vk)
+                            .collect::<Vec<_>>();
+                        let mut shielded = namada.shielded_mut().await;
+                        let _ = shielded.load().await;
+                        for vk in wallet_keys {
```

Comment:
Let's leave the internals of vk heights inside of the `sync` function. Instead lets concatenate all of the keys together and pass them into the shielded sync function.